src/Host.cpp

```diff
@@ -528,18 +528,18 @@ QString Host::getMmpMapLocation() const
     return mpMap->getMmpMapLocation();
 }
 
-bool Host::setDisplayFont(const QFont& font)
+std::pair<bool, QString> Host::setDisplayFont(const QFont& font)
 {
     const QFontMetrics metrics(font);
     if (metrics.averageCharWidth() == 0) {
-        return false;
+        return std::make_pair(false, QStringLiteral("specified font is invalid (its letters have 0 width)"));
```

Comment:
Hm, this is unfortunate. On the one hand, we may want to translate this when showing on the GUI, but we don't want to translate this when using it in Lua code...

Reply:
Indeed.

Reply:
To be honest showing "the selected font has 0 width characters" is going to be nonsense to the majority of the people. We can just say "this isn't a valid font" in the UI and they'll ignore it or ask for help for it.

Reply:
You're probably right.